src/Linter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace SLLH\ComposerLint;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class Linter
+{
+    /**
+     * @param array $manifest composer.json file manifest
+     *
+     * @return string[]
+     */
+    public function validate($manifest)
+    {
+        $errors = array();
+
+        if (isset($manifest['config']['sort-packages']) && $manifest['config']['sort-packages']) {
+            foreach (array('require', 'require-dev', 'conflict', 'replace', 'provide', 'suggest') as $linksSection) {
+                if (array_key_exists($linksSection, $manifest) && !$this->packagesAreSorted($manifest[$linksSection])) {
+                    array_push($errors, 'Links under '.$linksSection.' section are not sorted.');
+                }
+            }
+        }
+
+        return $errors;
+    }
+
+    private function packagesAreSorted(array $packages)
+    {
+        $names = array_keys($packages);
+
+        $hasPHP = in_array('php', $names);
+        $extNames = array_filter($names, function ($name) {
+            return 'ext-' === substr($name, 0, 4) && !strstr($name, '/');
```

Comment:
> This is certainly why the yoda condition is on the SF coding standards.

I not so sure they're always making the right decisions… worse, we've seen that the main reason they have for closing an RFC for a change in coding style is "this will cause merge nightmare", which means that even if they make a wrong decision and know it, they are going to stick with it forever. I don't think it makes sense to follow SF coding standard (some of them, but not all of them), especially in this package that is not an SF bundle.

> And I approve this. I prefer put yoda condition everywhere and be sure to have no error instead of always asking "May this will raise an error or not?" on each comparison instead of being concentrate on the logic itself.

This made some sense when there was no code review but now… this makes the code unclear (at least to coders that don't come from Dagobah) and although you free and yourself and the reviewer from the burden of checking if there is a risk for an assignment you did not really wish to make, you now make it harder for the reviewer and anyone (including you from the future) who reads this piece of code to understand it. I propose you continue to use Yoda condition everyone, and make the small effort of seeing if there is a risk or not, to see if it is really worth it. Because if there is a risk in 10% of the conditions, maybe this risk is worth taking, especially if you have unit tests and code review that will prevent you from doing this kind of mistake.

In my personal experience, I recall making this mistake some years ago. I still don't use Yoda conditions, and I really haven't been making this mistake for a long time. Not worth it IMO.